lib/vehicle.rb

```diff
@@ -1,18 +1,22 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
+  attr_reader   :vin,
+                :year,
+                :make,
+                :model,
+                :engine
+  attr_accessor :registration_date,
+                :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = nil
+    @plate_type = nil
```

Comment:
I would not have this as an instance variable. It is not dynamic this way and relies on another class to assign it. 
I would create a stand alone method called `plate_type` where you can the type based on the type of vehicle.